admin-console/src/main/webapp/src/components/EmailTemplates/InstantEmailTemplateModal.tsx

```diff
@@ -0,0 +1,82 @@
+import {
+    ActionGroup,
+    Button,
+    Form,
+    FormGroup,
+    FormSelect,
+    Modal,
+    ModalVariant
+} from '@patternfly/react-core';
+import React from 'react';
+
+import { InstantTemplate } from '../../types/Notifications';
+
+interface InstantTemplateModalProps {
+    showModal: boolean;
+    isEdit: boolean;
+    onClose: () => void;
+    templates: string[] | undefined;
+    initialInstantTemplate?: Partial<InstantTemplate>;
+    onSubmit: (instantTemplate: Partial<InstantTemplate>) => void;
```

Comment:
This could be changed to a non partial InstantTemplate, would need to validate and cast to non partial InstantTemplate
```suggestion
    onSubmit: (instantTemplate: InstantTemplate) => void;
```